tests/python/frontend/tflite/test_forward.py

```diff
@@ -0,0 +1,374 @@
+from tensorflow.python.ops import variables
+from tensorflow.contrib.lite.python import interpreter as interpreter_wrapper
+
+import nnvm.testing.tf
+
+
+
+#######################################################################
+# Generic run functions for TVM & TFLite
+# ------------------------------------------
+def convert_to_list(x):
+    if not isinstance(x, list):
+        x = [x]
+    return x
+
+def run_tvm_graph(tflite_model_buf, input_data, input_node, num_output=1, target='llvm',
+                  out_names=None):
+    """ Generic function to compile on relay and execute on tvm """
+    try:
+        import tflite.Model
+    except ImportError:
+        raise ImportError("The tflite package must be installed")
+
+    # get TFLite model from buffer
+    tflite_model = tflite.Model.Model.GetRootAsModel(tflite_model_buf, 0)
+
+    input_data = convert_to_list(input_data)
+    input_node = convert_to_list(input_node)
+
+    shape_dict = {}
+    dtype_dict = {}
+    for i, e in enumerate(input_node):
+        shape_dict[e] = input_data[i].shape
+        dtype_dict[e] = input_data[i].dtype.name
+
+    func, params = relay.frontend.from_tflite(tflite_model,
+                                              shape_dict=shape_dict,
+                                              dtype_dict=dtype_dict)
+    with relay.build_config(opt_level=3):
+        graph, lib, params = relay.build(func, target, params=params)
+
+    ctx = tvm.context(target, 0)
+    from tvm.contrib import graph_runtime
+    m = graph_runtime.create(graph, lib, ctx)
+    # set inputs
+    for i, e in enumerate(input_node):
+        m.set_input(e, tvm.nd.array(input_data[i].astype(input_data[i].dtype)))
+
+    m.set_input(**params)
+    # execute
+    m.run()
+    # get outputs
+    assert out_names is None or num_output == len(out_names), "out_names: {} num_output: {}".format(
+        out_names, num_output)
+    tvm_output_list = []
+    for i in range(0, num_output):
+        tvm_output = m.get_output(i)
+        tvm_output_list.append(tvm_output.asnumpy())
+    return tvm_output_list
+
+
+def run_tflite_graph(tflite_model_buf, input_data):
+    """ Generic function to execute TFLite """
+    input_data = convert_to_list(input_data)
+
+    interpreter = interpreter_wrapper.Interpreter(model_content=tflite_model_buf)
+    interpreter.allocate_tensors()
+
+    input_details = interpreter.get_input_details()
+    output_details = interpreter.get_output_details()
+
+    # set input
+    assert len(input_data) == len(input_details)
+    for i in range(len(input_details)):
+        interpreter.set_tensor(input_details[i]['index'], input_data[i])
+
+    # Run
+    interpreter.invoke()
+
+    # get output
+    tflite_output = list()
+    for i in range(len(output_details)):
+        tflite_output.append(interpreter.get_tensor(output_details[i]['index']))
+
+    return tflite_output
+
+
+def compare_tflite_with_tvm(tflite_in_data, tvm_in_data, in_name, input_tensors,
```

Comment:
Different input data for TVM and tflite ??
I understand this conversion is to felicitate NCHW across the model, this impose preprocessing.
Again non vision models may have challenges.

cc @tqchen  pls advice.

Reply:
This is discussed with @junrushao1994 before. As discussed, the issue is not our specific. Like TF-CoreML (https://github.com/tf-coreml/tf-coreml/blob/master/tfcoreml/_shape_sensitive_layers.py#L275), Intel OpenVINO, TensorRT and so on do the same things like us, we will meet the same problem. Yes, I agree there will be challenges, but currently let us start from little things, such as here limited 1-4D transformation and vision models (under internal development, we have tested many other models, doesn't have problems until now). When we want to do more generally and NLP, let us see how to handle it better.

Reply:
Yes. I tested Inception V3 just now(https://www.tensorflow.org/lite/models) and no problem, don't find official Inception V1 TFLite model. Maybe I can convert Tensorflow Inception V1 model into TFLite. Inception V3 model has one op named `concatation` not included in this PR, I will supply more ops later. If someone need Inception V3 support, I will provide `concation` op implementation without any problem. Current PR's main purpose is to provide infrastructure of TFLite frontend.